submissions/tavor118sn/task_friends_app/script.js

```diff
@@ -0,0 +1,156 @@
+// Global variables
+
+const CARD_LIST = document.getElementById('cardList');
+const FORM = document.forms.mainForm;
+const ASCENDING_ORDER = 'ascending';
+const DESCENDING_ORDER = 'descending';
+
+let friends;
+
+// URL
+
+const USER_FIELDS = ['gender', 'name', 'email', 'picture', 'dob'];
+
+const SITE_URL = 'https://randomuser.me/api/';
+const BASE_URL = new URL(SITE_URL);
+BASE_URL.searchParams.set('nat', 'us');
+BASE_URL.searchParams.set('results', '12');
+BASE_URL.searchParams.set('inc', USER_FIELDS.join());
+
+const buildUrl = (queryParam, paramValue) => {
+  const newUrl = new URL(BASE_URL);
+  if (queryParam && paramValue) {
+    newUrl.searchParams.set(queryParam, paramValue);
+  }
+  return newUrl;
+};
+
+// Render cards
+
+const createCard = (userInfo) => {
+  const userWholeCard = document.createElement('li');
+  userWholeCard.classList.add('user-card');
+
+  const userImg = document.createElement('img');
+  userImg.setAttribute('src', userInfo.picture.large);
+  userImg.setAttribute('alt', 'user_photo');
+  userImg.classList.add('user-photo');
+  userWholeCard.appendChild(userImg);
+
+  const cardHeader = document.createElement('h2');
+  cardHeader.classList.add('user-full-name');
+  cardHeader.innerHTML = `${userInfo.name.first} ${userInfo.name.last}`;
+  userWholeCard.appendChild(cardHeader);
+
+  const genderElem = document.createElement('p');
+  genderElem.innerHTML = `Gender: <strong>${userInfo.gender}</strong>`;
+  userWholeCard.appendChild(genderElem);
+
+  const ageElem = document.createElement('p');
+  ageElem.innerHTML = `Age: <strong>${userInfo.dob.age}</strong>`;
+  ageElem.classList.add('user-age');
+  userWholeCard.appendChild(ageElem);
+
+  return userWholeCard;
+};
+
+const generateFriendsListing = (friendList) => {
+  CARD_LIST.innerHTML = '';
+  const fragment = document.createDocumentFragment();
+
+  friendList.forEach((elem) => {
+    fragment.appendChild(createCard(elem));
+  });
+
+  CARD_LIST.appendChild(fragment);
+};
+
+// Load users
+
+async function loadUserList() {
+  const newUrl = buildUrl();
+  // try to make request at least 5 times in case of errors
+  for (let i = 0; i < 5; i++) {
+    try {
+      const response = await fetch(newUrl);
+
+      if (!response.ok) {
+        throw new Error(`HTTP error! status: ${response.status}`);
+      } else {
+        const friends = await response.json();
+        generateFriendsListing(friends.results);
```

Comment:
This function should be called not here.`loadUserList` loads data and rendering is out of scope for it.

Reply:
`generateFriendsListing` moved into initialization